editor/js/reaction.js

```diff
@@ -62,6 +62,9 @@ const session = {
 // Export session, because it's used by test.js.
 exports.session = session;
 
+const FLOAT_PATTERN = /^-?\d+(?:\.\d+)?(?:[eE]-?\d+)?$/;
```

Comment:
What about `^-?\d*(?:\.\d+)?(?:[eE]-?\d+)?$` to allow decimals w/o a leading 0? e.g., ".5" or "-.2"

Reply:
Hmm...that would also allow things like `e-14`. Expanded a little to make those cases mutually exclusive; PTAL.

Reply:
LGTM